vcstool/clients/vcs_base.py

```diff
@@ -132,3 +157,94 @@ def test_url(url, retry=2, retry_period=1, timeout=10):
                 timeout=timeout)
         raise URLError(str(e) + ' (%s)' % url)
     return response
+
+
+def _authenticated_urlopen(uri, timeout=None):
+    machine = urlparse(uri).netloc
+    if not machine:
+        return None
+
+    credentials = _credentials_for_machine(machine)
+    if credentials is None:
+        logger.warning('No credentials found for "%s"' % machine)
```

Comment:
I guess this poorly interacts with the progress dots, no? It also wouldn't appear in a repository specific output section which makes it difficult to figure out which repo is causing this. Maybe reporting this error case somehow differently would be better...

Reply:
Yeah it's not ideal, but I couldn't immediately think of a better way to do it. Shall I remove it for now? It will be fairly clear that authentication doesn't work because the pull will fail, so I'm not entirely sure how helpful this is.

Reply:
I've gone ahead and removed it in b49456971293af1141f9df1d0f051a32b4107fec.